### PR TITLE
[Fix] Implement ExonumJsonDeserialize for floats

### PR DESCRIPTION
**Pull request by vitvakatu**
This fix will allow to use `Vec<F32>` and `Vec<F64>` in `message!` macros.

### CODE REVIEW

File: exonum/src/encoding/float.rs [hunk 1]
```diff
@@ -243,6 +243,13 @@ impl ExonumJson for F32 {
     }
 }
 
+impl ExonumJsonDeserialize for F32 {
+    fn deserialize(value: &Value) -> Result<Self, Box<Error>> {
+        let number = value.as_f64().ok_or("Can't cast json as float")?;
+        Ok(Self::new(number as f32))
```
Comment (matklad) [inline exonum/src/encoding/float.rs]: This certainly should do some validation so as not to panic!?

Comment (vitvakatu) [reply exonum/src/encoding/float.rs]: Should we? afaik `as` will never panic, can you provide the example of the opposite?

Comment (matklad) [reply exonum/src/encoding/float.rs]: `new` panics for NaNs?

Comment (vitvakatu) [reply exonum/src/encoding/float.rs]: Nevermind, I see it now.

Comment (matklad) [reply exonum/src/encoding/float.rs]: Hm, even if the author of code forgets that new can panic, perhaps we should not have the new method at all? Or make it return an option?

Comment (vitvakatu) [reply exonum/src/encoding/float.rs]: :D The whole story about floats is a complete nightmare. But we need them anyway, and new method seems to be well documented. (how am I to read docs?).

**Pull request merged**